framework/wazuh/rbac/orm.py

```diff
@@ -80,16 +72,19 @@ class SecurityError(IntEnum):
     RELATIONSHIP_ERROR = -8
 
 
-@listens_for(_Session, 'after_flush')
```

Comment:
That whole code block was needed to clean empty relationships when one of the related resources was removed. It was added as a patch because the logic to do so when declaring the relationships was not working, but I found why and fixed it from scratch. This is an example:

```python
    roles = relationship("Roles", backref=backref("rules_associations", cascade="all,delete"))
```

We simply needed to add an actual `backref` object to the `backref` parameter to avoid interpreting this relationship as one of the parent. This way, it will properly consider this class and its relationships as children's and it will remove relationships on cascade when deleting one of the affected resources.

Reply:
Got it